### PR TITLE
Ll 3715 - wallet connect cli/poc

### PR DESCRIPTION
**Pull request by machard**
can try on https://example.walletconnect.org/

`ledger-live walletconnect -i 0 -c ethereum --walletConnectURI "QRCODEDATA"`
or 
`Ledger-live walletconnect -i 0 -c ethereum --walletConnectSession "sessionJSONString"`
sessionJSONString is provided in the log when using QRCODEDATA just under the line `connected`

### CODE REVIEW

File: cli/src/commands/walletconnect.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+      desc: "verbose mode",
+    },
+    {
+      name: "silent",
+      type: Boolean,
+      desc: "do not output the proxy logs",
+    },
+  ],
+  job: (opts: Opts) =>
+    Observable.create(async (o) => {
```
Comment (gre) [inline cli/src/commands/walletconnect.js:53]: minor note: Observable.create can't take an async function. the unsub returned value will not be used.
you could eventually put the logic out in an async function outside, which is going to be the case when things get stabilized back into live-common

File: cli/src/commands/walletconnect.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+                derivationMode: account.derivationMode,
+              };
+
+              log("walletconnect", "message to sign");
+              log("walletconnect", (message: any));
+
+              result = await withDevice(opts.device || "")((t) =>
+                from(signMessage(t, message))
+              )
+                .pipe(take(1))
```
Comment (gre) [inline cli/src/commands/walletconnect.js:139]: this is not needed. toPromise() implicitly is taking the first element already.

File: cli/src/commands/walletconnect.js [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                .toPromise();
+              result = result.signature;
+
+              log("walletconnect", "message signature");
+              log("walletconnect", result);
+
+              break;
+            case "eth_signTransaction":
+            case "eth_sendTransaction":
+              transactionRaw = payload.params[0];
```
Comment (gre) [inline cli/src/commands/walletconnect.js:149]: wording choice: probably use another name than "transactionRaw" to not confuse with the `TransactionRaw` type in live-common.

File: cli/src/commands/walletconnect.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+          log("walletconnect", approval);
+
+          connector.approveRequest(approval);
+        } else {
+          let message = "JSON RPC method not supported";
+          if (errorMsg) {
+            message = errorMsg;
+          }
+
+          const rejection: any = {
```
Comment (gre) [inline cli/src/commands/walletconnect.js:255]: casting something to `any` hides type error. to avoid.

**Review by gre — COMMENTED** [review]
nice progress

**Comment by machard** [comment]
I'm not sure I can put more stuff in common

**Pull request merged**